modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -338,8 +342,15 @@ protected CursorStreamProviderFactory getCursorStreamProviderFactory() {
     return empty();
   }
 
+  /**
+   * Returns a (if any) {@link ConfigurationModel} that is suitable for the {@link ComponentModel}. If there are more than one
+   * {@link ConfigurationModel} that can be used by this component, the ones that can be used implicitly (according to
+   * {@link ExtensionModelUtils#canBeUsedImplicitly(ParameterizedModel)}) will be prioritized.
+   */
   private Optional<ConfigurationModel> computeConfig() {
-    return getConfigurationForComponent(extensionModel, componentModel);
+    return getConfigurationForComponent(extensionModel, componentModel).stream().sorted(reverseOrder((x, y) -> Boolean
```

Comment:
That is the same as filter(is implicit).findFirst () only that way more inefficient. Change it